### PR TITLE
feat: move to typescript code generation

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
`docs-test` runs successfully locally, while fails in circleCI test, and it complains about the following links which can be clickable :(
```
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/delete_model.js,samples/README.md
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/get_dataset.js,samples/README.md
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/get_model_evaluation.js,samples/README.md
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/get_model.js,samples/README.md
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/get_operation_status.js,samples/README.md
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/export_dataset.js,samples/README.md
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/deploy_model.js,samples/README.md
```

**Comment by alexander-fenster** [comment]
@JustinBeckwith This is weird, they do fail on Actions:

```
[0] https://googleapis.dev/nodejs/automl/latest
[0] https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/googleapis/nodejs-automl&page=editor&open_in_editor=samples/deploy_model.js,samples/README.md
```

Would you be fine with ignoring those in `synth.py` and `linkinator.json` here and filing an issue for linkinator?

File: linkinator.config.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "skip": [
     "https://codecov.io/gh/googleapis/",
     "www.googleapis.com",
-    "setEndpoint.js",
-    "https://www.tensorflow.org/js",
-    "img.shields.io"
+    "img.shields.io",
+    "https://console.cloud.google.com/cloudshell/",
```
Comment (JustinBeckwith) [inline linkinator.config.json]: That ... seems like a problem.  Why do we need this url here?

File: linkinator.config.json [hunk 2]
```diff
@@ -3,8 +3,8 @@
...
+    "https://googleapis.dev/nodejs/automl/latest"
```
Comment (JustinBeckwith) [inline linkinator.config.json]: This exists though!

File: package.json [hunk 3]
```diff
@@ -8,10 +8,10 @@
     "node": ">=8.10.0"
   },
   "repository": "googleapis/nodejs-automl",
-  "main": "src/index.js",
+  "main": "build/src/index.js",
   "files": [
-    "protos",
-    "src"
+    "build/protos",
```
Comment (JustinBeckwith) [inline package.json]: Interesting.  @alexander-fenster in other places, we seem to import from just `"protos"`.  What's the difference?

File: synth.py [hunk 4]
```diff
@@ -19,20 +19,25 @@
...
+        extra_proto_files=['google/cloud/common_resources.proto'],
+    )
+    s.copy(library, excludes=['src/index.ts', 'README.md', 'package.json'])
 
 common_templates = gcp.CommonTemplates()
-templates = common_templates.node_library()
+templates = common_templates.node_library(source_location='build/src')
 s.copy(templates)
 
-s.replace('**/doc/google/protobuf/doc_timestamp.js',
```
Comment (JustinBeckwith) [inline synth.py]: So much nicer!

File: synth.py [hunk 5]
```diff
@@ -19,20 +19,25 @@
...
-        'https:\/\/cloud\.google\.com[\s\*]*http:\/\/(.*)[\s\*]*\)',
-        r"https://\1)")
-
 # Node.js specific cleanup
 subprocess.run(['npm', 'install'])
 subprocess.run(['npm', 'run', 'fix'])
```
Comment (JustinBeckwith) [inline synth.py]: Isn't there a `npx compileProtos src/` step or something like that which needs to go here?

File: package.json (hunk 3 above)
Comment (alexander-fenster) [reply package.json]: Are you talking about `.jsdoc.js` or `package.json`?  In `package.json`, it's always `build/protos` AFAIK.

**Comment by xiaozhenliu-gg5** [comment]
More links start to fail according to the test outcome:
```
[0] https://support.google.com/cloud/answer/6293499
[0] https://console.cloud.google.com/project
[0] https://console.cloud.google.com/flows/enableapi?apiid=automl.googleapis.com
[0] https://cloud.google.com/terms/launch-stages
```
but I reran the `docs-test` locally, no failure at all. Can I get some suggestion? @JustinBeckwith  @alexander-fenster It looks wired for me.

**Comment by xiaozhenliu-gg5** [comment]
Created an issue here: https://github.com/JustinBeckwith/linkinator/issues/156 if we want to clean the `synth.py` someday, this is for reference. Thank you!

**Comment by JustinBeckwith** [comment]
Awesome.  Go ahead and use `concurrency: 10` in the config.  I am going to send a hammer PR to update this everywhere so we don't run into trouble.

**Pull request merged**